avogadro/io/poscarformat.cpp

```diff
@@ -0,0 +1,337 @@
+/******************************************************************************
+
+  This source file is part of the Avogadro project.
+
+  Copyright 2016 Kitware, Inc.
+
+  This source code is released under the New BSD License, (the "License").
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+******************************************************************************/
+
+#include "poscarformat.h"
+
+#include <avogadro/core/elements.h> // for atomicNumberFromSymbol()
+#include <avogadro/core/matrix.h> // for matrix3
+#include <avogadro/core/molecule.h>
+#include <avogadro/core/vector.h> // for Vector3
+#include <avogadro/core/unitcell.h>
+#include <avogadro/core/utilities.h> // for split(), trimmed(), lexicalCast()
+
+#include <algorithm> // for std::count()
+#include <iomanip>
+#include <iostream>
+
+namespace Avogadro {
+namespace Io {
+
+using std::getline;
+using std::string;
+using std::vector;
+
+using Core::Array;
+using Core::Atom;
+using Core::Elements;
+using Core::Molecule;
+using Core::UnitCell;
+using Core::lexicalCast;
+using Core::split;
+using Core::trimmed;
+
+PoscarFormat::PoscarFormat()
+{
+}
+
+PoscarFormat::~PoscarFormat()
+{
+}
+
+bool PoscarFormat::read(std::istream &inStream, Core::Molecule &mol)
+{
+  size_t numLines = std::count(std::istreambuf_iterator<char>(inStream),
+                               std::istreambuf_iterator<char>(), '\n');
+
+  // There must be at least 7 "\n"'s to have a minimum crystal (including 1 atom)
+  if (numLines < 7) {
+    appendError("Error: POSCAR file is 7 or fewer lines long");
+    return false;
+  }
+
+  // We have to go back to the beginning if we are going to read again
+  inStream.clear();
+  inStream.seekg(0, std::ios::beg);
+
+  // We'll use these throughout
+  bool ok;
+  string line;
+  vector<string> stringSplit;
+
+  // First line is comment line
+  getline(inStream, line);
+  line = trimmed(line);
+  string title = " ";
+  if (!line.empty()) title = line;
```

Comment:
Please use a new line after the if statement,

if (!line.empty())
  title = line;